src/contracts/exchange/NiftyswapExchange20.sol

```diff
@@ -626,9 +697,54 @@ contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExch
    * @notice Prevents receiving Ether or calls to unsuported methods
    */
   fallback () external {
-    revert("NiftyswapExchange:UNSUPPORTED_METHOD");
+    revert("NiftyswapExchange20:UNSUPPORTED_METHOD");
   }
 
+  /***********************************|
+  |         Royalty Functions         |
+  |__________________________________*/
+
+  /**
+   * @notice Will set the royalties fees and recipient for contracts that don't support ERC-2981
+   * @param _fee       Fee pourcentage with a 1000 basis (e.g. 0.3% is 3 and 1% is 10 and 100% is 1000)
+   * @param _recipient Address where to send the fees to
+   */
+  function setRoyaltyInfo(uint256 _fee, address _recipient) onlyOwner public {
+    require(!IS_ERC2981, "NiftyswapExchange20#setRoyaltyInfo: TOKEN SUPPORTS ERC-2981");
+    require(_fee < FEE_MULTIPLIER, "NiftyswapExchange20#setRoyaltyInfo: ROYALTY_FEE_IS_TOO_HIGH");
```

Comment:
interesting, so if setting royalty directly, then the royalty fee must be lower than the LP fee..?

Reply:
Doesn't have to be, but this caps the total fee (LP + royalty) to a max of 100%, which is needed for when selling otherwise selling won't work since the marketplace will expect more fees than the value of the asset being sold.